packages/server/src/auth.ts

```diff
@@ -0,0 +1,31 @@
+import { PaymentMessageRequest } from "@celo/payments-types";
+import { buildTypedPaymentRequest } from "@celo/payments-utils";
+import { AddressUtils } from "@celo/utils";
+import { verifyEIP712TypedDataSigner } from "@celo/utils/lib/signatureUtils";
+import { kit } from "./services";
+
+export async function verifySignature(
+  signature: string,
+  account: string,
+  body: PaymentMessageRequest
+) {
+  try {
+    const accounts = await kit.contracts.getAccounts();
+    const dek = await accounts.getDataEncryptionKey(account);
+
+    const typedData = buildTypedPaymentRequest(
+      // @ts-ignore
```

Comment:
Why the ignore here?

Reply:
Something messed up with the typings

```
enum Enum {A, B, C};
type Type = {e: Enum.A, thing: 'A'} | {e: Enum.B, thing: 'B'} | {e: Enum.C, thing: 'C'} 
```

Then we switch on the `.e` property and it doesn't like it.